src/raft/raw_node.rs

```diff
@@ -300,6 +312,9 @@ impl<T: Storage> RawNode<T> {
            raft.raft_log.has_next_entries() {
             return true;
         }
+        if raft.read_state.index != INVALID_INDEX {
```

Comment:
@siddontang Any PR/progress? It's good that someone would handle this. But I could use a hand on it since I dislike waiting for some uncertain moment.